apps/rush-lib/src/logic/policy/ShrinkwrapFilePolicy.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as os from 'os';
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { IPolicyValidatorOptions } from './PolicyValidator';
+import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
+import { ShrinkwrapFileFactory } from '../ShrinkwrapFileFactory';
+
+/**
+ *  A policy that validates shrinkwrap files used by package managers.
+ */
+export class ShrinkwrapFilePolicy {
+  public static validate(rushConfiguration: RushConfiguration, options: IPolicyValidatorOptions): void {
+    // For now, we only have additional validation on PNPM shrinkwrap files
+    if (rushConfiguration.packageManager !== 'pnpm' || !rushConfiguration.pnpmOptions) {
```

Comment:
Is this a performance optimization? I assume this would otherwise just be covered by nothing happening on line 33.

Reply:
Yes this is just a performance optimization for the reason that there are no other shrinkwrap types that do validation, though since it's such a small perf hit to load the shrinkwrap, I think it might just be better to run it through just for the sake of consistent behaviour.